main.go

```diff
@@ -317,7 +317,7 @@ func getTracer(opts Options, out output) (opentracing.Tracer, io.Closer) {
 		tracer opentracing.Tracer = opentracing.NoopTracer{}
 		closer io.Closer
 	)
-	if opts.TOpts.Jaeger {
+	if opts.TOpts.Jaeger && !opts.TOpts.NoJaeger {
```

Comment:
Yeah this is an issue we have for all our flags -- I'm not sure what the right solution is for merging defaults and flags, but I think we should attack that in a separate PR.

For boolean flags specifically, I think if go-flags handled the `--no-FLAG` case, then it would help this quite a lot. The issue right now is that if a YAML file says "disable jaeger" (which is the default, so it would not really be a very useful option), then there's no way to enable Jaeger again.

This is more of a hacky solution so we can at least disable Jaeger, and then work on getting `--no-FLAG` into `go-flags` and that should be backwards compatible when we remove the flag here.